ducktape/tests/loader.py

```diff
@@ -446,31 +452,52 @@ def _is_test_function(self, function):
-            suites.extend(self._read_test_suite_from_file(test_suite_file_path))
+        
+        suites.extend(self._read_test_suite_from_file(test_suite_files))
 
         all_contexts = set()
         for suite in suites:
             all_contexts.update(self._load_test_suite(**suite))
         return all_contexts
 
-    def _read_test_suite_from_file(self, test_suite_file_path):
-        suites = list()
-        test_suite_file_path = os.path.abspath(test_suite_file_path)
-        if not os.path.exists(test_suite_file_path) or not os.path.isfile(test_suite_file_path):
-            raise LoaderException("Test suite does not exist or is a directory: " + test_suite_file_path)
-
-        with open(test_suite_file_path) as fp:
+    def _load_suite(self, suite_file_path):
+        with open(suite_file_path) as fp:
             try:
                 file_content = yaml.load(fp, Loader=yaml.FullLoader)
             except Exception as e:
-                raise LoaderException("Failed to load test suite from file: " + test_suite_file_path, e)
-
-            if not file_content:
-                raise LoaderException("Test suite file is empty: " + test_suite_file_path)
-            if not isinstance(file_content, dict):
-                raise LoaderException("Malformed test suite file: " + test_suite_file_path)
-
+                raise LoaderException("Failed to load test suite from file: " + suite_file_path, e)
+
+        if not file_content:
+            raise LoaderException("Test suite file is empty: " + suite_file_path)
+        if not isinstance(file_content, dict):
+            raise LoaderException("Malformed test suite file: " + suite_file_path)
+
+        for suite_name, suite_content in file_content.items():
+            if not suite_content:
+                raise LoaderException("Empty test suite " + suite_name + " in " + suite_file_path)
+        return file_content
+
+    def _read_test_suite_from_file(self, root_suite_file_paths):
+        suites = []
+        root_suite_file_paths = [os.path.abspath(file_path) for file_path in root_suite_file_paths]
+        files = {file: self._load_suite(file) for file in root_suite_file_paths}
+        stack = root_suite_file_paths
+
+        while len(stack) != 0:
+            curr = stack.pop()
+            loaded = files[curr]
+            if 'import' in loaded:
+                directory = os.path.dirname(curr)
+                # apply path of current file to the files inside
+                abs_file_iter = (os.path.abspath(os.path.join(directory, file))
+                                    for file in loaded.get('import', []))
+                imported = [file for file in abs_file_iter if file not in files]
+                for file in imported:
+                    files[file] = self._load_suite(file)
+                stack.extend(imported)
+                del files[curr]['import']
+
+        for test_suite_file_path, file_content in files.items():
```

Comment:
You could probably extract everything from line 501 onward into a separate method to make testing easier, but I think all our unit tests load full suite files anyway, so not a big deal.

```python
    for test_suite_file_path, file_content in files.items():
        suites.extend(_load_suite(test_suite_file_path, file_content))

# rename existing _load_suite into _load_file since it doesn't actually load suite, only a full file content.
def _load_suite(file_path, file_content):
    # everything from line 501:
    for suite_name, suite_content in file_content.items():
        .....
```